Source/JavaScriptCore/wasm/WasmOperationsInlines.h

```diff
@@ -292,6 +300,8 @@ inline EncodedJSValue arrayNewElem(JSWebAssemblyInstance* instance, uint32_t typ
     JSGlobalObject* globalObject = instance->globalObject();
     VM& vm = globalObject->vm();
     auto* array = JSWebAssemblyArray::create(vm, globalObject->webAssemblyArrayStructure(), FieldType { storageType, Mutability::Mutable }, arraySize, WTFMove(arrayRTT));
+    if (!array)
+        return JSValue::encode(jsNull());
```

Comment:
Is returning null correct? What is the type of wasm instruction we are expecting?

Reply:
We're expecting `array.new_elem` from the GC extension [spec](https://webassembly.github.io/gc/core/appendix/index-instructions.html#index-instr). AFAICT our callers convert an `isNull()` result to `Wasm::ExceptionType::BadArrayNewInitElem`, for which the associated message is "Out of bounds or failed to allocate in array.new_elem", so `jsNull()` seems appropriate here (I guess if we were drawing the distinction, this would be a "failed to allocate").